### PR TITLE
Add eslint and stylelint and remove jshint and jscs

### PR DESCRIPTION
**Pull request by amire80**
The first commit adds eslint and disables many rules.

Further commits gradually enable rules and fix the issues.

Some rules that are less trivial to fix are still disabled intentionally, but will be fixed later once CI is more stable.

### CODE REVIEW

**Comment by kartikm** [comment]
Please squash all commits into single commit. See: http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

File: Gruntfile.js [hunk 1]
```diff
@@ -65,45 +63,32 @@ module.exports = function ( grunt ) {
...
 		},
-		jscs: {
-			fix: {
-				options: {
-					fix: true
-				},
-				src: '<%= jshint.all %>'
+		stylelint: {
+			options: {
+				syntax: 'css'
```
Comment (kartikm) [inline Gruntfile.js]: less instead of css!

File: Gruntfile.js [hunk 2]
```diff
@@ -65,45 +63,32 @@ module.exports = function ( grunt ) {
...
-			main: {
-				src: '<%= jshint.all %>'
-			}
-		},
-		csslint: {
-			all: [
-				'css/**/*.css'
+			src: [
+				'**/*.css',
+				'!node_modules/**'
```
Comment (kartikm) [inline Gruntfile.js]: Also ignore dist, libs here.

File: Gruntfile.js (hunk 1 above)
Comment (amire80) [reply Gruntfile.js]: Done, thank you!

File: Gruntfile.js (hunk 2 above)
Comment (amire80) [reply Gruntfile.js]: Done, thank you!

**Comment by kartikm** [comment]
Please rebase on the top of latest commit!

**Comment by kartikm** [comment]
Ping @amire80 for rebase+squash!

**Comment by kartikm** [comment]
Deprecated by #547

**Pull request closed without merging**